src/renderer/components/app.tsx

```diff
@@ -74,11 +74,11 @@ export class App extends React.Component {
   }
 
   renderExtensionRoutes() {
-    return clusterPageRegistry.getItems().map(({ id: pageId, components: { Page }, exact, routePath, subPages }) => {
+    return clusterPageRegistry.getItems().map(({ components: { Page }, exact, routePath, subPages }) => {
       const Component = () => {
         if (subPages) {
           const tabs: TabLayoutRoute[] = subPages.map(({ exact, routePath, components: { Page } }) => {
-            const menuItem = clusterPageMenuRegistry.getById(pageId);
+            const menuItem = clusterPageMenuRegistry.getByRoutePath(routePath);
```

Comment:
You can't match page like this when it has defined `routePath: "/extra-path/:someId"` for example

Reply:
So please return matching by ID

Reply:
Matching by `id` was only working within an extension. Introduced "target" concept that can work also across extensions.